edge-hub/src/Microsoft.Azure.Devices.Routing.Core/RouteResult.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Routing.Core
+{
+    using System;
+
+    public class RouteResult : IEquatable<RouteResult>

```

Comment:
I noticed that a lot of classes implement IEquatable interface, but I think it is a pretty strong constraint for classes. https://rules.sonarsource.com/csharp/RSPEC-4035 

@varunpuranik Could you please elaborate it little bit about reasons to have them?

Reply:
The reason is mainly to have a strongly typed way to check equality to another object of the same type. I think most of these classes can be made sealed..

Reply:
I only added it to facilitate updating existing test collateral, where Endpoint comparison now becomes RouteResult comparison.  It's not actually used anywhere in product code, but updating all our tests without IEquatable here would be too expensive.

---
In reply to: [376152050](https://github.com/Azure/iotedge/pull/2469#discussion_r376152050) [](ancestors = 376152050)